srml/staking/src/lib.rs

```diff
@@ -1148,15 +1195,15 @@ impl<T: Trait> Module<T> {
 			let validators = elected_stashes.into_iter()
 				.map(|s| Self::bonded(s).unwrap_or_default())
 				.collect::<Vec<_>>();
-			(slot_stake, Some(validators))
```

Comment:
I think slot_stake is not taken care of correctly. So far in my skim of the code:
- it looks like it is not deprecated and plays a major role in computing reward: https://github.com/paritytech/substrate/pull/2882/files#diff-b5a024b4f4ae96b0bfc0681acae56709R1046
- It is not returned from `select_validators` and not updated inside the function. 
- I don't see any other changes pointing to it.

If you can pass the staking tests and my guess is correct then I will highly doubt the value of these tests :p

Reply:
It is populated there https://github.com/paritytech/substrate/blob/bf6504458de76e8e28b9c96371ae0465212fdb6d/srml/staking/src/lib.rs#L1183-L1191

But that said indeed I still have to update test